src/invidious/helpers/youtube_api.cr

```diff
@@ -20,12 +35,86 @@ def request_youtube_api_browse(continuation)
     "continuation": continuation,
   }
 
-  # Send the POST request and return result
+  return _youtube_api_post_json("/youtubei/v1/browse", data)
+end
+
+def request_youtube_api_browse(browse_id : String, params : String)
+  # JSON Request data, required by the API
+  data = {
+    "browseId" => browse_id,
+    "context"  => {
+      "client" => {
+        "hl"            => "en",
+        "gl"            => "US",
+        "clientName"    => "WEB",
+        "clientVersion" => HARDCODED_CLIENT_VERS,
+      },
+    },
+  }
+
+  # Append the additionnal parameters if those were provided
+  # (this is required for channel info, playlist and community, e.g)
+  if params != ""
+    data["params"] = params
+  end
+
+  return _youtube_api_post_json("/youtubei/v1/browse", data)
+end
+
+####################################################################
+# request_youtube_api_search(search_query, params, region)
+#
+# Requests the youtubei/vi/search endpoint with the required headers
+# to get JSON in en-US (english US).
+#
+# The requested data is a search string, with some additional
+# paramters, formatted as a base64 string.
+#
+def request_youtube_api_search(search_query : String, params : String, region = nil)
+  # JSON Request data, required by the API
+  data = {
+    "query":   URI.encode_www_form(search_query),
+    "context": {
+      "client": {
+        "hl":            "en",
+        "gl":            region || "US", # Can't be empty!
```

Comment:
If you want to ignore the warning about the value can't be a nil you can do it this way instead:
```json
"gl": region.not_nil!
```

Reply:
Youtube will return a 400 error code is you don't provide valid `hl` and `gl` parameters, so `region` CAN'T be `nil`